source/include/logging/cs_Logger.h

```diff
@@ -46,323 +47,647 @@
-		cs_log_end();
-	}
+#undef _logArray
+#define _logArray(level, addNewLine, pointer, size, ...)
+#endif
 
-	// This function takes an unused argument "fmt".
-	// This fmt ends up in the .ii file (preprocessed .cpp file), and is then used to gather all log strings.
-	// To make it easier for the compiler to optimize out the fmt string, we only call cs_log_args without the fmt arg here.
-	template<class... Args>
-	void cs_log_args(uint32_t fileNameHash, uint32_t lineNumber, uint8_t logLevel, bool addNewLine, const char* fmt, const Args&... args) {
-		cs_log_args_no_fmt(fileNameHash, lineNumber, logLevel, addNewLine, args...);
+// Write a string with printf functionality.
+#ifdef HOST_TARGET
+__attribute__((unused)) static bool _logPrefixHost = true;
+
+#define _FILE (sizeof(__FILE__) > 30 ? __FILE__ + (sizeof(__FILE__) - 30 - 1) : __FILE__)
+
+#undef _log
+#define _log(level, addNewLine, fmt, ...)                  \
+	if (level <= SERIAL_VERBOSITY) {                       \
+		if (_logPrefixHost) {                              \
+			printf("[%-30.30s : %-4d] ", _FILE, __LINE__); \
+		}                                                  \
+		printf(fmt, ##__VA_ARGS__);                        \
+		if (addNewLine) {                                  \
+			printf("\r\n");                                \
+		}                                                  \
+		_logPrefixHost = addNewLine;                       \
 	}
 
-	// Write logs as plain text.
-	#if CS_UART_BINARY_PROTOCOL_ENABLED == 0
-		void cs_log_printf(const char *str, ...);
-		__attribute__((unused)) static bool _logPrefixPlainText = true;
-
-		// Adding the file name and line number, adds a lot to the binary size.
-		#define _FILE (sizeof(__FILE__) > 30 ? __FILE__ + (sizeof(__FILE__)-30-1) : __FILE__)
-
-		#undef _log
-		#define _log(level, addNewLine, fmt, ...) \
-				if (level <= SERIAL_VERBOSITY) { \
-					if (_logPrefixPlainText) { \
-						cs_log_printf("[%-30.30s : %-4d] ", _FILE, __LINE__); \
-					} \
-					cs_log_printf(fmt, ##__VA_ARGS__); \
-					if (addNewLine) { \
-						cs_log_printf("\r\n"); \
-					} \
-					_logPrefixPlainText = addNewLine; \
-				}
-
-		#undef _logArray
-		#define _logArray(level, addNewLine, pointer, size, ...)
-	#endif
-
-	// Write a string with printf functionality.
-	#ifdef HOST_TARGET
-		__attribute__((unused)) static bool _logPrefixHost = true;
-
-		#define _FILE (sizeof(__FILE__) > 30 ? __FILE__ + (sizeof(__FILE__)-30-1) : __FILE__)
-
-		#undef _log
-		#define _log(level, addNewLine, fmt, ...) \
-				if (level <= SERIAL_VERBOSITY) { \
-					if (_logPrefixHost) { \
-						printf("[%-30.30s : %-4d] ", _FILE, __LINE__); \
-					} \
-					printf(fmt, ##__VA_ARGS__); \
-					if (addNewLine) { \
-						printf("\r\n"); \
-					} \
-					_logPrefixHost = addNewLine; \
-				}
-
-		#undef _logArray
-		#define _logArray(level, addNewLine, pointer, size, ...)
-	#endif
-
-
+#undef _logArray
+#define _logArray(level, addNewLine, pointer, size, ...)
 #endif
```

Comment:
This used to be better readable imo.

Reply:
This is anyway an unreadable file. If this needs to be addressed it should be split over multiple files.